linfa-linear/src/glm.rs

```diff
@@ -0,0 +1,355 @@
+            return Err(LinearError::InvalidValue(format!(
+                "Some value(s) of y are out of the valid range for power value {}",
+                self.power
+            )));
+        }
+
+        // We initialize the coefficients and intercept
+        let mut coef: Array1<A>;
+        if self.fit_intercept {
+            coef = Array::zeros(x.ncols() + 1);
+            let temp = *link.link(&array![y.mean().unwrap()]).get(0).unwrap();
+            let element = coef.get_mut(0).unwrap();
+            *element = temp;
+        } else {
+            coef = Array::zeros(x.ncols());
+        }
+
+        // We optimize the parameters using LBFGS solver
+        let problem = TweedieProblem {
+            x,
+            y,
+            fit_intercept: self.fit_intercept,
+            link: &link,
+            dist,
+            alpha: self.alpha,
+        };
+        let linesearch = MoreThuenteLineSearch::new();
+        let solver = LBFGS::new(linesearch, 7).with_tol_grad(A::from(self.tol).unwrap());
+        let result = Executor::new(problem, solver, ArgminParam(coef))
+            .max_iters(self.max_iter as u64)
+            .run()?;
+        coef = result.state.get_best_param().as_array().to_owned();
+
+        if self.fit_intercept {
+            Ok(FittedTweedieRegressor {
+                coef: coef.slice(s![1..]).to_owned(),
+                intercept: *coef.get(0).unwrap(),
+                link,
+            })
+        } else {
+            Ok(FittedTweedieRegressor {
+                coef: coef.to_owned(),
+                intercept: A::from(0.).unwrap(),
+                link,
+            })
+        }
+    }
+}
+
+struct TweedieProblem<'a, A: Float> {
+    x: &'a Array2<A>,
+    y: &'a Array1<A>,
+    fit_intercept: bool,
+    link: &'a Link,
+    dist: TweedieDistribution,
+    alpha: f64,
+}
+
+impl<'a, A: Float> TweedieProblem<'a, A> {
+    fn ypred(&self, p: &Array1<A>) -> (Array1<A>, Array1<A>, usize) {
+        let mut offset = 0;
+        let mut intercept = A::from(0.).unwrap();
+        if self.fit_intercept {
+            offset = 1;
+            intercept = *p.get(0).unwrap();
+        }
+
+        let lin_pred = self
+            .x
+            .view()
+            .dot(&p.slice(s![offset..]))
+            .mapv(|x| x + intercept);
+
+        (self.link.inverse(&lin_pred), lin_pred, offset)
+    }
+}
+
+impl<'a, A: Float> ArgminOp for TweedieProblem<'a, A> {
+    type Param = ArgminParam<A>;
+    type Output = A;
+    type Hessian = ();
+    type Jacobian = Array1<A>;
+    type Float = A;
+
+    fn apply(&self, p: &Self::Param) -> std::result::Result<Self::Output, argmin::core::Error> {
+        let p = p.as_array();
+
+        let (ypred, _, offset) = self.ypred(&p);
+
+        let dev = self.dist.deviance(self.y, &ypred)?;
+
+        let pscaled = p
+            .slice(s![offset..])
+            .mapv(|x| x * A::from(self.alpha).unwrap());
+
+        let obj = A::from(0.5).unwrap() * dev
```

Comment:
can you add a comment here, where this comes from? You can leave out one of the `0.5` here :)

Reply:
I made the change with the multiplier.
I have added a comment on the `apply` function, which encapsulates the objective function we are trying to minimize using LBFGS.